crates/vm/levm/src/opcode_handlers/push.rs

```diff
@@ -25,20 +26,34 @@ impl VM {
             .checked_add(1)
             .ok_or(VMError::InvalidOpcode)?;
 
-        let next_n_bytes = current_call_frame
+        let read_n_bytes: Vec<u8> = current_call_frame
             .bytecode
-            .get(
-                current_call_frame.pc()
-                    ..current_call_frame
-                        .pc()
-                        .checked_add(n_bytes)
-                        .ok_or(VMError::Internal(InternalError::PCOverflowed))?,
-            )
-            .ok_or(VMError::InvalidBytecode)?; // This shouldn't happen during execution
+            .get(current_call_frame.pc()..)
+            .unwrap_or_default()
+            .iter()
+            .take(n_bytes)
+            .cloned()
+            .collect();
+
+        let mut value_to_push = [0; WORD_SIZE];
 
-        let value_to_push = U256::from(next_n_bytes);
+        let start_index = WORD_SIZE.checked_sub(n_bytes).ok_or(VMError::Internal(
+            InternalError::ArithmeticOperationUnderflow,
+        ))?;
+        value_to_push
+            .get_mut(
+                start_index
+                    ..start_index
+                        .checked_add(read_n_bytes.len())
+                        .ok_or(VMError::Internal(
+                            InternalError::ArithmeticOperationOverflow,
+                        ))?,
+            )
+            .unwrap_or_default()
+            .copy_from_slice(&read_n_bytes);
```

Comment:
`copy_from_slice` can panic so maybe we shouldn't use it. It looks better though.

Reply:
Done! In [Avoid using copy_from_slice, since can panic](https://github.com/lambdaclass/lambda_ethereum_rust/pull/1195/commits/8cc2a1b041990f91270a5ccdf6651983b0dea79c)